pandas/core/arrays/categorical.py

```diff
@@ -1740,6 +1740,12 @@ def take_nd(self, indexer, allow_fill=True, fill_value=None):
         # but is passed thru internally
         assert isna(fill_value)
 
+        indexer = np.asarray(indexer)
+
+        # take on empty array only valid if result is an all-missing array
+        if not len(self) and not (indexer == -1).all():
```

Comment:
this should be in ``take_1d``

Reply:
pls make this change

Reply:
Did you see my answer/question the first time you asked this? 

I gave reasons not to do this, so at least please answer to them.

Reply:
I already responsded. ``take_1d`` can certainly be used in an implementation of this. This is the general implementation of virtually all take behavior, so naturally a check for no indexers should happen there, rather than in each higher level implementation (e.g. EA). The point is that the way you are advocating is simply more complex / buggy and adds more code. Virtually all EA methods should use API's that pandas already has. Should these be public. Sure if possible. But for something like this we *don't* want to make the actual implementation public as it has certain guarantees. However I don't see any problem with an internal implementation (like EA) using it. it fact I would say they *have* / *should* use it. otherwise how else would categorical/DTI/II be implemented at all?